### DISCUSSION TITLE
The images are stitched longitudinally error

### DISCUSSION THREAD

**Discussion by qiuwulun795** [OpenStitching/stitching#209] [Q&A]
I have two images that I have stitched horizontally, and I want to stitch them vertically, but they get the following error:
Traceback (most recent call last):
  File "D:\anaconda3\envs\stitch_github\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "D:\anaconda3\envs\stitch_github\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\anaconda3\envs\stitch_github\Scripts\stitch.exe\__main__.py", line 7, in <module>
  File "D:\anaconda3\envs\stitch_github\lib\site-packages\stitching\cli\stitch.py", line 312, in main
    panorama = stitcher.stitch(images, feature_masks)
  File "D:\anaconda3\envs\stitch_github\lib\site-packages\stitching\stitcher.py", line 102, in stitch
    cameras = self.refine_camera_parameters(features, matches, cameras)
  File "D:\anaconda3\envs\stitch_github\lib\site-packages\stitching\stitcher.py", line 157, in refine_camera_parameters
    return self.camera_adjuster.adjust(features, matches, cameras)
  File "D:\anaconda3\envs\stitch_github\lib\site-packages\stitching\camera_adjuster.py", line 48, in adjust
    raise StitchingError("Camera parameters adjusting failed.")
stitching.stitching_error.StitchingError: Camera parameters adjusting failed.
![32](https://github.com/OpenStitching/stitching/assets/108721267/2f42ea55-67ae-4814-9b93-1ab798241f10)
![33](https://github.com/OpenStitching/stitching/assets/108721267/9e4b14fd-76b6-443c-a7c9-6b2520c3fec4)
These are my two stitched images. I think their overlap rate is still very high. How can I solve this problem，Looking forward to your reply，thanks

**Comment by lukasalexanderweber** [comment]
maybe use `--wave_correct_kind vert`. but maybe the black borders will harm the algorithm. Can't you just stitch all images in one step?

**Reply by qiuwulun795** [reply]
I followed your instructions and still reported the same error。
he reason why I don't stitch all the pictures at once is that my project is to take stitching pictures in real time. First, I take five pictures of the inner wall of the tunnel and splice them into a horizontal pattern. Then, I take another five pictures and splice them into a horizontal pattern. But now I'm having a lot of trouble with vertical stitching。

**Reply by lukasalexanderweber** [reply]
How do you aquire your Images? Can you make a Sketch?

**Reply by qiuwulun795** [reply]
In the horizontal stitching, I have a handheld device with five cameras distributed in a fan shape. These cameras simultaneously capture patterns and perform horizontal stitching. The results are as shown in the above two pictures. The complete tunnel expansion diagram can be obtained by concatenating them longitudinally. What I am doing now is vertical concatenation, and the problem I encounter is also related to the vertical concatenation of the horizontal expansion diagram。
I wanted to use open source projects directly to get vertical stitching results, but that didn't seem to work。Thanks again for your reply
